nano/node/transport/tcp_listener.cpp

```diff
@@ -300,12 +300,38 @@ auto nano::transport::tcp_listener::check_limits (boost::asio::ip::address const
 	return accept_result::accepted;
 }
 
-std::size_t nano::transport::tcp_listener::connection_count () const
+size_t nano::transport::tcp_listener::connection_count () const
 {
 	nano::lock_guard<nano::mutex> lock{ mutex };
 	return connections.size ();
 }
 
+size_t nano::transport::tcp_listener::realtime_count () const
```

Comment:
This is an O(1) -> O(N) tradeoff. However, I think the encapsulation benefits exceed the cost.

Reply:
Given that N is bounded (by default to something like 1000 I think) the benefits of having a clear, predicate based logic outweigh any performance concerns. And this is nowhere near the node hot path anyway.